Empire/Empire.h

```diff
@@ -316,6 +317,14 @@ class FO_COMMON_API Empire {
     int                     CapitalID() const;                          ///< Returns the numeric ID of the empire's capital
     int                     StockpileID(ResourceType res) const;        ///< Returns the numeric ID of the empire's stockpile location for the resource of type \a res
 
+    /** Return an object id that is owned by the empire or
+        INVALID_OBJECT_ID. SourceID() is safe to call with a null this
+        pointer. */
+    int                     SourceID() const;
```

Comment:
calling a member function on a null pointer is probably undefined behaviour... as above, no.